Src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs

```diff
@@ -231,7 +231,7 @@ protected virtual SyntaxNode TransformReducedNode(SyntaxNode reducedNode, Syntax
             }, cancellationToken);
             }
 
-            Task.WaitAll(simplifyTasks, cancellationToken);
```

Comment:
Oh.... i think i see.  Task.WhenAll will only complete once all the inner tasks are done.  You're trying to get it so that the task transitions immediately to cancelled if the cancellationToken triggers.

That's definitely interesting.  Though is it necessary in practice?  Are our inner reduction tasks not responding to cancellation quickly enough?